### DISCUSSION TITLE
Eeprom library fail after soft reset

### DISCUSSION THREAD

**Discussion by Endorfin35** [stm32duino/Arduino_Core_STM32#2293] [Libraries] [answered]
Hello everyone. I am coding on STM32G030C6 with Platformio and arduino core.

I am using the current version of stm32duino. I can save data to flash with the eeprom library and read it without any problems.

After I cut off the power and when I re-energize it, I can access the eeprom records. Again, no problem here.

Here's the problem; HAL_NVIC_SystemReset(); If I restart with the function, eprom records are read incorrectly. I get wrong values on the first few readings. Then I start reading the correct values. But it seems completely inconsistent. To overcome this problem, I read with a loop and ignored them. Then I read again and tried to reach the real values, but I got inconsistent results. Sometimes it worked, sometimes it didn't...

I'm pretty newbie at this. What mistake could I be doing?

**Comment by fpistm** [comment]
Well probably the FLASH unlock requires some times after this kind of reset.
If the FLASH unlock fails, the STM32 eeproom simply do nothing. That's why you didn't get the value you want.
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/9f8f2593-f2dd-4e83-bbcd-b2204b8ce844)

Maybe the BSY1 bit is set and you must wait.

**Comment by Endorfin35** [comment, marked as answer]
I tried again at home after work. It works smoothly. I'm absolutely confused. I'll check it for a few days for reliability.

Thanks.
